### ISSUE TITLE
Midterm Feedback

### ISSUE THREAD

**Issue by kmn5409** [frankiscuwu/cs506-final-project#4]
In terms of your CNN, depending on the size of the dataset dropout might not be necessary if a dataset is too small, dropout is generally used to help with generalization, but if the dataset is too much the model might not be able to learn enough if you are dropping important information. Another thing to consider is making the learning rate small, you can drop values of 0.1 or even 0.05. I’m not really sure why you tried to implement the same CNN’s using Tensorflow and Pytorch, they are mainly just tools you use to implement Neural Networks, there maybe some randomness involved in building one or the other, but there is no reason to implement the same architecture on both libraries. Additionally the default parameters for Tensorflow and Pytorch might be slightly different which could account for the slight differences in accuracies. You should also be looking at additional metrics like Precision, Recall and F1-score since this is a multi-classification problem, since the classes may be imbalanced so accuracy may be somewhat misleading. Please refer to the previous Github issues because we did highlight some of these in the past.

I am unclear as to what your next steps are for the project. Do you plan to implement any other models? Some other simple models would be to maybe utilize a simple model like ResNet depending on how long it could train, or you could look into how you could use SVM’s for this problem. Additionally there are not many visualizations of the data or exploration of the data. You could look at using SVD for visualizing your data and look at the data with respect to the classes to see if you see any interesting patterns. I don’t even really see an exploration of the class counts for the training data really, just more for the predictions, some visualizations of the distribution would be good to have. Remember the project for this course wants to see you looking at data exploration and data visualizations, having models is good as well, but is not the primary focus.

Additionally, looking at techniques like Grad-CAM might be useful to understand which regions of an input image are important for the network’s prediction of a particular class. Also consider trying to implement other data augmentation techniques even if it’s simpler ones at least to get experience using it and is part of the project outline.
